### ISSUE TITLE
Error sending event on compile time failure

### ISSUE THREAD

**Issue by DanielRedOak** [DataDog/chef-handler-datadog#92]
If there is a compile time error (in my case, CTRL-C right after dd-handler initialized) chef throws an error: [2016-12-08T20:54:15+00:00] ERROR: Report handler Chef::Handler::Datadog raised #<NoMethodError: undefined method `length' for nil:NilClass> 

This is the line in question:  https://github.com/DataDog/chef-handler-datadog/blob/master/lib/chef/handler/datadog_chef_events.rb#L140

The problem is that elasped_time is not nil and  @run_status.all_resources is nil since it hasn't collected everything yet so .length on nil throws the error.  Here is a snippet of what is available at this time:

```
@updated_resources=1, @up_to_date_resources=6, @successful_audits=0, @failed_audits=0, @start_time=2016-12-08 20:53:57 +0000, @end_time=2016-12-08 20:53:57 +0000, @skipped_resources=0, @current_recipe="datadog::dd-handler">
```

**Comment by olivielpeau** [comment]
Hi @DanielRedOak and thanks for reporting this issue!

I think this could be fixed by using in `datadog_chef_events.rb` the logic that's used for the metrics to detect whether there was a compile error: see https://github.com/DataDog/chef-handler-datadog/blob/v0.10.0/lib/chef/handler/datadog_chef_metrics.rb#L52-L54

We could maybe factorize this logic directly in `lib/chef/handler/datadog.rb` to avoid the duplication.

Feel free to open a PR fixing this, I'm adding this issue to the next minor milestone to make sure we take care of it if you can't.

**Comment by miketheman** [comment]
See this old PR on the compile time topic. https://github.com/DataDog/chef-handler-datadog/pull/55#issuecomment-68407536

**Comment by olivielpeau** [comment]
Fixed in #96, thanks @DanielRedOak for your detailed description of the issue!
